controllers/persistentvolumeclaim_controller_test.go

```diff
@@ -0,0 +1,95 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/topolvm/topolvm"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("PersistentVolumeClaimController controller", func() {
+
+	ctx := context.Background()
+	var stopFunc func()
+
+	BeforeEach(func() {
+		err := k8sClient.DeleteAllOf(ctx, &corev1.PersistentVolumeClaim{}, client.InNamespace("test"))
+		Expect(err).NotTo(HaveOccurred())
+
+		time.Sleep(100 * time.Millisecond)
+
+		mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+			Scheme: scheme,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		reconciler := PersistentVolumeClaimReconciler{
+			Client: k8sClient,
+		}
+		err = reconciler.SetupWithManager(mgr)
+		Expect(err).NotTo(HaveOccurred())
+
+		ctx, cancel := context.WithCancel(ctx)
+		stopFunc = cancel
+		go func() {
+			err := mgr.Start(ctx)
```

Comment:
To stop it properly, why don't you use a chan to signal its stop?

Reply:
Though I tried to fix it, I'm not confident with my understainding.
Are the changes in the following commit what you intended?

7d4eb1bd6721551f46d11cb1fe36df12578c188f

Reply:
I mean use a chan to transfer err value.
For example:

```
var errCh chan err
BeforeEach(func() {
  ...
  errCh <- mgr.Start(ctx)
})

AfterEach(func() {
  stopFunc()
  <- errCh
   ...
})
```
WaitGroup is also good.

BTW, why do you use `for` to start mgr?

Reply:
Thank you for your explanation. I got it.
I modified the code in a similar way to what you showed.